middleware/changeLanguage.js

```diff
@@ -28,84 +27,34 @@ example response from language web service:
 }
 **/
 
-function setup() {
-  var transport = service.findById();
-  var middleware = function(req, res, next) {
-
-    // no-op, request did not require a language change
-    if( !isLanguageChangeRequired( req, res ) ){
+function setup(service, should_execute) {
+  return function controller(req, res, next) {
+    if (!should_execute(req, res)) {
       return next();
     }
 
-    // collect a list of parent ids to fetch translations for
-    var ids = extractIds( res );
-
-    // perform language lookup for all relevant ids
-    var timer = (new Date()).getTime();
-    transport.query( ids, function( err, translations ){
-
-      // update documents using a translation map
-      if( err ){
-        logger.error( '[language] [error]', err );
-      } else {
-        updateDocs( req, res, translations );
+    service(req, res, (err, translations) => {
+      // if there's an error, log it and bail
+      if (err) {
+        logger.info(`[middleware:language][error]`);
+        logger.error(err);
+        return next();
```

Comment:
Is this a good place to pass the error to next () like we talked about with lily this morning? Not urgent but since we were on the subject.